src/main/scala/tilelink/Broadcast.scala

```diff
@@ -159,119 +184,214 @@ class TLBroadcast(lineBytes: Int, numTrackers: Int = 4, bufferless: Boolean = fa
-          TLMessages.Hint           -> MuxLookup(in.a.bits.param, Wire(UInt(width = 2)), Array(
-            TLHints.PREFETCH_READ   -> TLPermissions.toB,
-            TLHints.PREFETCH_WRITE  -> TLPermissions.toN)),
-          TLMessages.AcquireBlock   -> acq_perms,
-          TLMessages.AcquirePerm    -> acq_perms))
+      val acq_needT = MuxLookup(in.a.bits.param, Wire(Bool()), Array(
+        TLPermissions.NtoB -> false.B,
+        TLPermissions.NtoT -> true.B,
+        TLPermissions.BtoT -> true.B))
+      val req_needT = MuxLookup(in.a.bits.opcode, Wire(Bool()), Array(
+        TLMessages.PutFullData    -> true.B,
+        TLMessages.PutPartialData -> true.B,
+        TLMessages.ArithmeticData -> true.B,
+        TLMessages.LogicalData    -> true.B,
+        TLMessages.Get            -> false.B,
+        TLMessages.Hint           -> MuxLookup(in.a.bits.param, Wire(Bool()), Array(
+          TLHints.PREFETCH_READ   -> false.B,
+          TLHints.PREFETCH_WRITE  -> true.B)),
+        TLMessages.AcquireBlock   -> acq_needT,
+        TLMessages.AcquirePerm    -> acq_needT))
+
+      filter.io.request.valid := in.a.valid && a_first && trackerReady
+      filter.io.request.bits.mshr    := OHToUInt(selectTracker)
+      filter.io.request.bits.address := in.a.bits.address >> lineShift
+      filter.io.request.bits.needT   := req_needT
+      filter.io.request.bits.allocOH := a_cache // Note: this assumes a cache doing MMIO has allocated
+
+      val leaveB = !filter.io.response.bits.needT && !filter.io.response.bits.gaveT
+      val others = filter.io.response.bits.cacheOH & ~filter.io.response.bits.allocOH
+      val todo = Mux(leaveB, 0.U, others)
+      filter.io.response.ready := !probe_busy
+      when (filter.io.response.fire()) {
+        probe_todo  := todo
+        probe_line  := filter.io.response.bits.address
+        probe_perms := Mux(filter.io.response.bits.needT, TLPermissions.toN, TLPermissions.toB)
+      }
+
+      // Inform the tracker of the number of probe targets
+      val responseCache = filter.io.response.bits.cacheOH | filter.io.response.bits.allocOH
+      val responseCount = PopCount(todo)
+      val responseMSHR = UIntToOH(filter.io.response.bits.mshr, numTrackers).asBools
+      val sack = filter.io.response.fire() && leaveB && others =/= 0.U
+      (trackers zip responseMSHR) foreach { case (tracker, select) =>
+        tracker.probe.valid := filter.io.response.fire() && select
+        tracker.probe.bits.count   := responseCount
+        tracker.probe.bits.cacheOH := responseCache
+        when (sack && select) { tracker.probesack := true.B }
       }
 
       // The outer TL connections may not be cached
-      out.b.ready := Bool(true)
-      out.c.valid := Bool(false)
-      out.e.valid := Bool(false)
+      out.b.ready := true.B
+      out.c.valid := false.B
+      out.e.valid := false.B
     }
   }
 }
 
+// addressBits is after removing the block offset (typically lowest 6 bits for 64 bytes)
```

Comment:
isn't this a "tag"?

Reply:
Well, you would have to strip sector bits and row bits as well before it becomes a tag. I agree addressBits is a bad name. I could not think of a better one, though. If you have a suggestion, I'll change it.

Reply:
`blockAddressBits` or `lineAddressBits` maybe?

Reply:
Changed to blockAddressBits.